posts/exploring-iot-infrastructure-support-in-aws-part2.md

```diff
@@ -0,0 +1,450 @@
+
+This lead me down the path of trying to figure out if it was possible to use the AWS Python SDK on MicroPython instead. I had a feeling that MicroPython was somewhat limited when it came to larger libraries and would not have enough storage. `boto3` is large and would likely suffocate the memory on the device.
+
+> Due to ESP8266 RAM limitations, the python modules for this project need to be included in the flashed firmware in frozen bytecode
+
+There also appears to be issues with the mutual [TLS](https://github.com/micropython/micropython/issues/2781). It looks like MicroPython is useful but it is probably not suitable for this AWS proof of concept. Perhaps there is something better for AWS integration.
+
+# MongooseOS - The King
+
+After leaving MicroPython, I came across `MongooseOS` mentioned in some of the AWS blog [posts](https://aws.amazon.com/blogs/apn/aws-iot-on-mongoose-os-part-1/). The posts provided useful information that the AWS documentation does not necessarily mention. I thought I would explore it more.
+
+It appears MongooseOS and the ESP devices work very nicely together. Mongoose provides a browser IDE with Cloud building, logging and automatic configuration setup. There is no need for repackaging or modification of source for TLS. MongooseOS also provides a abstract implementation of IoT for AWS, Azure and GCP.
+
+I thought I would give it go so I [downloaded](https://mongoose-os.com/software.html) the MongooseOS tool called `mos` and ran the IDE to see if it really lived up to the name.
+
+Once MongooseOS was installed all I needed to do was start the `mos` browser IDE, download the sample project through the web interface then flash it to the device. Once flashed, I had the option of setting up WiFi access by loading the credentials onto the device filesystem.
+
+![MOS tool setup](https://raw.githubusercontent.com/BenEdridge/blog-usa/master/images/2018/07/mos_initial_setup_screenshot.png)
+
+Once the initial setup is finished the device rebooted and console output was routed directly to the Web logging output. MongooseOS provides an unified cloud library and specialised [AWS libraries](https://mongoose-os.com/docs/cloud/aws.md) to interact with the AWS device shadows. We can subscribe to messages and shadow updates if needed.
+
+To communicate with our IoT devices we will be using MQTT, this allows us to access the default MQTT topics for Shadow updates and any custom topics created. We will create 2 extra topic for publishing sensor data and device settings to AWS as follows: `devices/<deviceId>/data` and `devices/<deviceId>/settings`
+
+We will be logging humidity and temperature at selected intervals using MQTT. Then Setting up actions on AWS IoT to save this data to a S3 bucket.
+
+## MongooseOS was loaded with the following code in our `init.js` file:
+
+## Device State and Settings:
+
+```JavaScript
```

Comment:
- You've introduced JavaScript here with no description about how this is supported in Mongoose.
- You might want to look at having a diagram to explain this. There is no context around the code.